### DISCUSSION TITLE
Vdatum requirements for mesh generation

### DISCUSSION THREAD

**Discussion by saeed-moghimi-noaa** [noaa-ocs-modeling/OCSMesh#152] [Ideas]
## Vdatum requirements for mesh generation
@LiujuanRachelTang (please tag Elena to this discussion board)
 @SorooshMani-NOAA @felicio93 

I start this discussion board to gather mesh generation requirements for Vdatum mesh generation efforts.

From what I have got from Elena's presentation today:

- Most important feature for the mesh generation package is the Mesh-size function capability

What she did not like is
- the mesh generation package does not check if the inlet is closed or not!?  @LiujuanRachelTang  Please elaborate

Thanks

**Comment by saeed-moghimi-noaa** [comment]
@Delta-function
Hi Elena,

Please help documenting what you like and dislike from the mesh generation tools you use (see my text above). We can work on this in a meeting as well.

I mainly looking for technical capabilities and perhaps the limits that you use to set the size function for instance:

- size function
- * M2 tide creteria!?
- * distance from a shape-file /  contour line
- * slope / depth dependent mesh size
- * !?
- missing features:
- * check closure of a inlet
-  * !?




Thanks,
-Saeed

**Reply by Delta-function** [reply]
Hi Saeed,

Automatically (and soundly) generated size function is indeed a very useful feature of OceanMesh2D. There are a number of parameters to manipulate the size function - for their detailed description, I'd probably better refer you to their user guide https://www3.nd.edu/~coast/reports_papers/2018-oceanmesh2d-user-guide.pdf

We do set a number of points per M2 wavelength (coupled with min/max for offshore) which are the defining parameters offshore, min/max spacings on the coastline which define edge size by the coast (coupled with the coast curvature calculated by OceanMesh), and the width of the nearshore which defines the width of the transition zone from one size function to the other. OceanMesh2D generates very nice triangles with a gradual transition between the resolutions. It also allows connecting/embedding areas with different size function parameters. 

A missing feature that I mentioned was the lack of checking that the shoreline does not cross a mesh element. So they would not make a triangle of nodes located on opposite sides of a jetty or a barrier island. I'm not sure what "closure of an inlet" is about.

Thanks,
Elena

**Comment by LiujuanRachelTang** [comment]
Hi Saeed,

Please add Elena to the discussion. She is VDatum's technical lead.
I am not sure how to tag her. Thanks.


-Rachel

**Reply by Delta-function** [reply]
Thanks, Rachel, I was in the discussion already. It's just that GitHub uses my gmail address, which I haven't checked until the weekend.

**Comment by saeed-moghimi-noaa** [comment]
Hi @LiujuanRachelTang 

Elena's ( @Delta-function ) github account was already there. I did not notice it when I originally tried to tag her.

Thanks,

**Comment by saeed-moghimi-noaa** [comment]
@Delta-function 

Thanks for the explanation. Lets continue this conversation. The inlet closure was my not very accurate understanding of your point. Thanks for the correction.

Best.
